### ISSUE TITLE
Emscripten web build freezes/crashes when entering battle mode

### ISSUE THREAD

**Issue by bob-the-hamster** [ohrrpgce/ohrrpgce#1268]
Build the latest nightly wip using emscripten
Play an rpg file with battles (I used testgame/turntest.rpg)
Enter a battle, the game will freeze/lock or show an error
(I saw errors a week or two ago, seems to freeze every time for me now)

**Comment by bob-the-hamster** [comment]
After firefox warns that the script is slowing down the browser, and I kill it, this is the log message in the js console

```
Script terminated by timeout at:
ohrrpgce-game.wasm.dlmalloc@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[5376]:0x3e93f3
instrumentWasmExports/ret[x]@http://localhost:8000/ohrrpgce-game.js:9454:33
allocateData@http://localhost:8000/ohrrpgce-game.js:9527:26
handleSleep@http://localhost:8000/ohrrpgce-game.js:9617:42
_emscripten_sleep@http://localhost:8000/ohrrpgce-game.js:9235:23
ohrrpgce-game.wasm.Emscripten_GLES_SwapWindow@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[4336]:0x33e9a4
ohrrpgce-game.wasm.SDL_GL_SwapWindow@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[4452]:0x34d5f5
ohrrpgce-game.wasm.GLES2_RenderPresent@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[4076]:0x30be50
ohrrpgce-game.wasm.SDL_RenderPresent@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[4050]:0x3056e7
ohrrpgce-game.wasm.present_internal2@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[2041]:0x1c6351
ohrrpgce-game.wasm.gfx_sdl2_setpal@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[2044]:0x1c8667
ohrrpgce-game.wasm.MAYBE_DO_GFX_SETPAL@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[2195]:0x1df20f
ohrrpgce-game.wasm.FADETOPAL_INTERNAL@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[2196]:0x1df9b3
ohrrpgce-game.wasm.FADETOCOLOR@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[2202]:0x1e0ac7
ohrrpgce-game.wasm.FADEOUT(long, long)@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[595]:0x82423
ohrrpgce-game.wasm.BATTLE@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[261]:0x7fc0
ohrrpgce-game.wasm.UPDATE_HEROES@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[1616]:0x144733
ohrrpgce-game.wasm.main@http://localhost:8000/ohrrpgce-game.wasm:wasm-function[1714]:0x160237
instrumentWasmExports/ret[x]@http://localhost:8000/ohrrpgce-game.js:9454:33
doRewind@http://localhost:8000/ohrrpgce-game.js:9552:16
handleSleep/<@http://localhost:8000/ohrrpgce-game.js:9578:47
callUserCallback@http://localhost:8000/ohrrpgce-game.js:4903:9
safeSetTimeout/<@http://localhost:8000/ohrrpgce-game.js:4917:25
```

**Comment by bob-the-hamster** [comment]
Fixed in commit 05090dc27d19da -- thanks to Blain Maguire
